pkg/kinflate/transformers/overlay_test.go

```diff
@@ -149,6 +150,277 @@ func TestOverlayRun(t *testing.T) {
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			Data: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "apps/v1",
+					"kind":       "Deployment",
+					"metadata": map[string]interface{}{
+						"name": "deploy1",
+					},
+					"spec": map[string]interface{}{
+						"template": map[string]interface{}{
+							"spec": map[string]interface{}{
+								"containers": []interface{}{
+									map[string]interface{}{
+										"name": "nginx",
+										"env": []interface{}{
+											map[string]interface{}{
+												"name":  "ANOTHERENV",
+												"value": "HELLO",
+											},
+										},
+									},
+									map[string]interface{}{
+										"name":  "busybox",
+										"image": "busybox",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	expected := resource.ResourceCollection{
+		{
+			GVK:  schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "Deployment"},
+			Name: "deploy1",
+		}: &resource.Resource{
+			Data: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "apps/v1",
+					"kind":       "Deployment",
+					"metadata": map[string]interface{}{
+						"name": "deploy1",
+					},
+					"spec": map[string]interface{}{
+						"template": map[string]interface{}{
+							"metadata": map[string]interface{}{
+								"labels": map[string]interface{}{
+									"old-label":     "old-value",
+									"another-label": "foo",
+								},
+							},
+							"spec": map[string]interface{}{
+								"containers": []interface{}{
+									map[string]interface{}{
+										"name":  "nginx",
+										"image": "nginx:latest",
+										"env": []interface{}{
+											map[string]interface{}{
+												"name":  "ANOTHERENV",
+												"value": "HELLO",
+											},
+											map[string]interface{}{
+												"name":  "SOMEENV",
+												"value": "BAR",
+											},
+										},
+									},
+									map[string]interface{}{
+										"name":  "busybox",
+										"image": "busybox",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	lt, err := NewOverlayTransformer(overlay)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	err = lt.Transform(base)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if !reflect.DeepEqual(base, expected) {
+		err = compareMap(base, expected)
+		t.Fatalf("actual doesn't match expected: %v", err)
+	}
+}
+
+func TestMultiplePatchesWithConflict(t *testing.T) {
+	base := resource.ResourceCollection{
+		{
+			GVK:  schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "Deployment"},
+			Name: "deploy1",
+		}: &resource.Resource{
+			Data: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "apps/v1",
+					"kind":       "Deployment",
+					"metadata": map[string]interface{}{
+						"name": "deploy1",
+					},
+					"spec": map[string]interface{}{
```

Comment:
Can we use much simpler examples maybe? Much more to the point?

Reply:
Will try to simplify this.

Reply:
Simplified it a bit.